src/en/readallcomicscom/src/eu/kanade/tachiyomi/extension/en/readallcomicscom/ReadAllComics.kt

```
@@ -0,0 +1,193 @@
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import org.jsoup.select.Elements
+import rx.Observable
+
+class ReadAllComics : ParsedHttpSource() {
+
+    override val name = "ReadAllComics"
+
+    override val baseUrl = "https://readallcomics.com"
+
+    override val lang = "en"
+
+    override val supportsLatest = false
+
+    private lateinit var searchPageElements: Elements
+
+    override val client = network.cloudflareClient.newBuilder()
+        .addInterceptor(::archivedCategoryInterceptor)
+        .rateLimit(2)
+        .build()
+
+    private fun archivedCategoryInterceptor(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+
+        val document = Jsoup.parse(
+            response.peekBody(Long.MAX_VALUE).string(),
+            request.url.toString(),
+        )
+
+        val newUrl = document.selectFirst(".description-archive > p > span > a")
+            ?.attr("href")?.toHttpUrlOrNull()
+            ?: return response
+
+        if (newUrl.pathSegments.contains("category")) {
+            response.close()
+
+            return chain.proceed(
+                request.newBuilder()
+                    .url(newUrl)
+                    .build(),
+            )
+        }
+
+        return response
+    }
+
+    override fun popularMangaRequest(page: Int) =
+        GET("$baseUrl${if (page > 1)"/page/$page/" else ""}", headers)
```

This page does not seem to work.

---

It's true, this does not work at the moment because readallcomics doesn't really have a functional homepage at present. I asked about this in the [discord programming channel](https://discord.com/channels/1193460528052453448/1195207602200649850/1365845921941684244) and was advised by a maintainer that removing it to throw an error and leaving it (so functionality picks back up if the homepage is restored by the source) were both acceptable.

I've chosen the latter because it may not require another change if and when that homepage is restored.

---

Reading from their disclaimer, they intended to do it that way (they declare themselves as a search engine). Better to throw an exception instead as said on the channel.